components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java

```diff
@@ -3123,7 +3141,9 @@ public static void addUser(UserInfoBean userInfoBean) throws RestAPIException {
        try {
             StratosUserManagerUtils.addUser(getTenantUserStoreManager(), userInfoBean);
         } catch (UserManagerException e) {
-            throw new RestAPIException(e.getMessage());
+           String msg = "Error in adding User";
+           log.error(msg, e);
+           throw new RestAPIException(msg, e);
         }
```

Comment:
Better to have more descriptive error message.You can use some user specific details with error message.